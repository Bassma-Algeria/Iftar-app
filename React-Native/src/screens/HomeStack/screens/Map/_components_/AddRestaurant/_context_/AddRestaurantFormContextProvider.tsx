import React, {createContext, useState} from 'react';
import type {RestaurantInfoToAdd} from '../../../../../../../Gateways/RestaurantsGateway/RestaurantsGateway.interface';

interface AddRestaurantContextValues {
  restaurantInfo: RestaurantInfoToAdd;
  setRestaurantInfo: React.Dispatch<React.SetStateAction<RestaurantInfoToAdd>>;

  serverError: boolean;
  setServerError: React.Dispatch<React.SetStateAction<boolean>>;
}

const AddRestaurantFormContext = createContext<AddRestaurantContextValues | undefined>(undefined);

const AddRestaurantFormContextProvider: React.FC = ({children}) => {
  const [serverError, setServerError] = useState<boolean>(false);
  const [restaurantInfo, setRestaurantInfo] = useState<RestaurantInfoToAdd>({
    name: '',
    ownerName: '',
    pictures: [],
    closingTime: '',
    openingTime: '',
    locationCoords: undefined,
    placeName: '',
  });

  return (
    <AddRestaurantFormContext.Provider
      value={{
        restaurantInfo,
        setRestaurantInfo,
        serverError,
        setServerError,
      }}>
      {children}
    </AddRestaurantFormContext.Provider>
  );
};

export {AddRestaurantFormContextProvider, AddRestaurantFormContext};
