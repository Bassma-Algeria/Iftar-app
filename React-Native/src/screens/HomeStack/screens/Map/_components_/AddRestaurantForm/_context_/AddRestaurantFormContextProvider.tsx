import React, {createContext, useState} from 'react';
import type {RestaurantInfoToAdd} from '../../../../../../../Gateways/RestaurantsGateway/RestaurantsGateway.interface';

interface IRestaurantForm extends Omit<RestaurantInfoToAdd, 'pictures'> {
  pictures: {uri: string; base64: string}[];
}

interface AddRestaurantContextValues {
  restaurantInfo: IRestaurantForm;
  setRestaurantInfo: React.Dispatch<React.SetStateAction<IRestaurantForm>>;

  serverError: boolean;
  setServerError: React.Dispatch<React.SetStateAction<boolean>>;
}

const AddRestaurantFormContext = createContext<AddRestaurantContextValues | undefined>(undefined);

const initialAddRestaurantFormState: IRestaurantForm = {
  name: '',
  locationName: '',
  locationCoords: {latitude: 0, longitude: 0},
  workingTime: {opening: {hour: 0, minut: 0}, closing: {hour: 0, minut: 0}},
  pictures: [],
};

const AddRestaurantFormContextProvider: React.FC = ({children}) => {
  const [serverError, setServerError] = useState<boolean>(false);
  const [restaurantInfo, setRestaurantInfo] = useState<IRestaurantForm>(
    initialAddRestaurantFormState,
  );

  return (
    <AddRestaurantFormContext.Provider
      value={{restaurantInfo, setRestaurantInfo, serverError, setServerError}}>
      {children}
    </AddRestaurantFormContext.Provider>
  );
};

export {AddRestaurantFormContextProvider, AddRestaurantFormContext, initialAddRestaurantFormState};
