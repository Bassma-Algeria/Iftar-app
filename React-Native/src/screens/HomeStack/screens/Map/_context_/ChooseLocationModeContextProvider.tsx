import React, {createContext, useState} from 'react';

import type {LocationCords} from '../../../../../@types/LocationCords';

type OnConfirmFunction = (selectedLocation: LocationCords) => void;

interface ChooseLocationModeContextValues {
  selectedLocation?: LocationCords;
  setSelectedLocation: React.Dispatch<React.SetStateAction<LocationCords | undefined>>;

  onConfirm?: OnConfirmFunction;
  setOnConfirm: React.Dispatch<React.SetStateAction<OnConfirmFunction | undefined>>;
}

export const ChooseLocationModeContext = createContext<ChooseLocationModeContextValues | undefined>(
  undefined,
);

const ChooseLocationModeContextProvider: React.FC = ({children}) => {
  const [selectedLocation, setSelectedLocation] = useState<LocationCords>();
  const [onConfirm, setOnConfirm] = useState<OnConfirmFunction>();

  return (
    <ChooseLocationModeContext.Provider
      value={{selectedLocation, setSelectedLocation, onConfirm, setOnConfirm}}>
      {children}
    </ChooseLocationModeContext.Provider>
  );
};

export {ChooseLocationModeContextProvider};
