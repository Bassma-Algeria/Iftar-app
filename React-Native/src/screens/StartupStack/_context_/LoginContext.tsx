import React, {createContext, useState} from 'react';
import {LoginInfo} from '../../../Gateways/AuthGateway/AuthGateway.interface';

interface LoginContextValues {
  loginInfo: LoginInfo;
  setLoginInfo: React.Dispatch<React.SetStateAction<LoginInfo>>;
}

export const LoginContext = createContext<LoginContextValues | undefined>(undefined);

const LoginContextProvider: React.FC = ({children}) => {
  const [loginInfo, setLoginInfo] = useState<LoginInfo>({email: '', password: ''});
  return (
    <LoginContext.Provider
      value={{
        loginInfo,
        setLoginInfo,
      }}>
      {children}
    </LoginContext.Provider>
  );
};

export {LoginContextProvider};
