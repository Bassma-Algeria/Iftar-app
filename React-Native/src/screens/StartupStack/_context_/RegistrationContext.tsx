import React, {createContext, useState} from 'react';
import {SignupInfo} from '../../../Gateways/AuthGateway/AuthGateway.interface';

interface RegistrationContextValues {
  registrationInfo: SignupInfo;
  setRegistrationInfo: React.Dispatch<React.SetStateAction<SignupInfo>>;
}

export const RegistrationContext = createContext<RegistrationContextValues | undefined>(undefined);

const RegistrationContextProvider: React.FC = ({children}) => {
  const [registrationInfo, setRegistrationInfo] = useState<SignupInfo>({
    email: '',
    password: '',
    confirmPassword: '',
    phoneNumber: '',
  });
  return (
    <RegistrationContext.Provider
      value={{
        registrationInfo,
        setRegistrationInfo,
      }}>
      {children}
    </RegistrationContext.Provider>
  );
};

export {RegistrationContextProvider};
