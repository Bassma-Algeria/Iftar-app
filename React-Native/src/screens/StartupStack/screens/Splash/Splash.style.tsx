import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    paddingTop: 100,
  },
});

export {styles};
