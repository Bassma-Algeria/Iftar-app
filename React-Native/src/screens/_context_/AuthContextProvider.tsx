import React, {createContext, useState} from 'react';

interface AuthContextValues {
  isRestaurantOwner: boolean;
  setIsRestaurantOwner: React.Dispatch<React.SetStateAction<boolean>>;
}

const AuthContext = createContext<AuthContextValues | undefined>(undefined);

const AuthContextProvider: React.FC = ({children}) => {
  const [isRestaurantOwner, setIsRestaurantOwner] = useState<boolean>(false);

  return (
    <AuthContext.Provider value={{isRestaurantOwner, setIsRestaurantOwner}}>
      {children}
    </AuthContext.Provider>
  );
};

export {AuthContextProvider, AuthContext};
