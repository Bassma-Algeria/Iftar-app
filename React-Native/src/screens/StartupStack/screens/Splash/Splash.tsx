import React, {useCallback, useEffect} from 'react';
import {Text, View} from 'react-native';

import {styles} from './Splash.style';

import {MainStackScreenProps} from '../../../MainStack.types';

import {useAuthContext} from '../../../_hooks_/useAuthContext';
import {localStorage} from '../../../../utils/helpers/LocalStorage';
import {Layout} from '../../_components_/Layout/Layout';
import {Logo} from '../../../../components/Logo/Logo';

interface Props extends MainStackScreenProps<'StartupStack'> {}

const Splash: React.FC<Props> = ({navigation}) => {
  const {setIsRestaurantOwner} = useAuthContext();

  const setup = useCallback(async () => {
    const isFirstTime = await localStorage.get('firstTime');
    if (!isFirstTime) {
      return navigation.replace('StartupStack', {screen: 'Welcom'});
    }

    const token = await localStorage.get('token');
    if (!token) {
      return navigation.replace('StartupStack', {screen: 'ChooseUsageType'});
    }

    setIsRestaurantOwner(true);
    navigation.navigate('HomeStack', {screen: 'Map'});
  }, [navigation, setIsRestaurantOwner]);

  useEffect(() => {
    setup();
  }, [setup]);

  return (
    <Layout>
      <View style={styles.container}>
        <Logo />
      </View>
    </Layout>
  );
};

export {Splash};
