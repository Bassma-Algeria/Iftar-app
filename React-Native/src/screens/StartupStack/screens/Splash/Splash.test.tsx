import React from 'react';
import renderer from 'react-test-renderer';
import {AuthContextProvider} from '../../../_context_/AuthContextProvider';

import {Splash} from './Splash';

describe('Splash Screen', () => {
  it('should render correctly', () => {
    const props: any = {navigation: {navigate: jest.fn(), replace: jest.fn()}};
    const tree = renderer
      .create(
        <AuthContextProvider>
          <Splash {...props} />
        </AuthContextProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
