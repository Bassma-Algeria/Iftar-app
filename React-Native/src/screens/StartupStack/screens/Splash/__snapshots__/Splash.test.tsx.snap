// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash Screen should render correctly 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "backgroundColor": "#F4E6CA",
        "height": "100%",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "position": "absolute",
          "top": -30,
          "width": "100%",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "maxWidth": 200,
          }
        }
      />
      <Image
        resizeMode="contain"
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "maxWidth": 200,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginTop": "60%",
          "padding": 15,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "paddingTop": 100,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "baseline",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#7C4908",
                    "fontFamily": "Cairo-Bold",
                    "fontSize": 35,
                  },
                  undefined,
                ]
              }
            >
              إفطار
            </Text>
            <Image
              collapsable={false}
              nativeID="animatedComponent"
              resizeMode="contain"
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 22,
                  "left": 3,
                  "position": "absolute",
                  "top": -5,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                  "width": 22,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
